src/views/AddItem.jsx

```diff
@@ -1,3 +1,108 @@
+import { useState, useEffect } from 'react';
+import { addItem } from '../api/firebase';
+
+const dayConverter = (text) => {
+	if (text === 'soon') {
+		return 7;
+	} else if (text === 'kind-of-soon') {
+		return 14;
+	} else {
+		return 30;
+	}
+};
 export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+	const [itemName, setItemName] = useState('');
+	const [frequency, setFrequency] = useState('soon');
+	const [message, setMessage] = useState('');
+	const listId = 'my test list';
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		const daysUntilNextPurchase = dayConverter(frequency);
+		try {
+			await addItem(listId, { itemName, daysUntilNextPurchase });
+			setMessage(`${itemName} was added to the list`);
+			setItemName('');
+			setFrequency('soon');
+		} catch (err) {
+			console.error(err);
+			setMessage(`Failed to Add: ${itemName}`);
+		}
+	};
+	const handleChange = (e) => {
```

Comment:
**Suggestion/Nit:** You could consider renaming `handleChange()` to something more descriptive, to help other developers better understand what this function is impacting.

Since this is specifically handling the _frequency_ change and never the text input change, how about `handleFrequencyChange()`?

Reply:
Done, thanks for the suggestion. I really like that you are thinking of code readability.